DNN Platform/Library/Entities/Users/UserController.cs

```diff
@@ -692,7 +692,9 @@ public static void ApproveUser(UserInfo user)
         public static void UserAgreedToTerms(UserInfo user)
         {
             Requires.NotNull("user", user);
-            MembershipProvider.Instance().UserAgreedToTerms(user);
+            var u = user;
+            u.PortalID = PortalController.GetEffectivePortalId(u.PortalID);
+            MembershipProvider.Instance().UserAgreedToTerms(u);

```

Comment:
`var u = user;` doesn't have any effect here.  Since `UserInfo` is a class, you're still pointing to the shared instance, and this `PortalID` assignment will affect the `UserInfo` object that's passed in.  I think we'd want to avoid that, so it's probably better to either pass `portalId` as a parameter to `UserAgreedToTerms` or check the effective portal ID inside `UserAgreedToTerms`.

Reply:
OK, I'll need to rewrite that bit of the API though, which I wanted to avoid.

Reply:
Is it possible to move the `GetEffectivePortalId` call into `UserAgreedToTerms`?  That seems like the least intrusive change, if there's nothing standing in the way…

Reply:
I'll give it a shot. Hang on.

Reply:
Like it is now?

Reply:
yup, looks good to me!